### DISCUSSION TITLE
Github Actions on user's Github Pages site (user.github.io)

### DISCUSSION THREAD

**Discussion by janreggie** [community/community#22841] [Other Feature Feedback, Questions, & Ideas]
<p>Hello. I would like to ask for some help regarding using GitHub Actions on a user’s GitHub Pages site (in my case, <a href="https://github.com/janreggie/janreggie.github.io/" rel="noopener nofollow ugc">https://github.com/janreggie/janreggie.github.io/</a>). I have the following file on <code>.github/workflows/deploy.yml</code>:</p>
<pre><code class="lang-auto">name: deploy

on:
  push:
    branches:
      - main  # Set a branch to deploy

jobs:
  deploy:
    runs-on: ubuntu-18.04
    steps:
      - uses: actions/checkout@v2
        with:
          submodules: true  # Fetch Hugo themes (true OR recursive)
          fetch-depth: 0    # Fetch all history for .GitInfo and .Lastmod

      - name: Setup Hugo
        uses: peaceiris/actions-hugo@v2
        with:
          hugo-version: '0.75.1'
          extended: true

      - name: Build
        run: hugo --minify

      - name: Deploy
        uses: peaceiris/actions-gh-pages@v3
        with:
          github_token: ${{ secrets.GITHUB_TOKEN }}
          publish_dir: ./public
          publish_branch: gh-pages
</code></pre>
<p>This should perform some action when I push anything to <code>master</code> (and the file is indeed present in <code>master</code>) such that it compiles the website and pushes to <code>gh-pages</code>. However, whenever I push, the Actions tab is empty:</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/9/9e69329102f38af1467b291e958a66475b5807a0.png" data-download-href="/uploads/short-url/mBmP9g22SPKgF1J6pBaxXZBZIkg.png?dl=1" title="Screenshot_2020-09-17 janreggie janreggie github io" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/33337903/181073352-23c0c164-b6ea-4274-b000-9fab909fdd6f.png" alt="Screenshot_2020-09-17 janreggie janreggie github io" data-base62-sha1="mBmP9g22SPKgF1J6pBaxXZBZIkg" width="690" height="336" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/9/9e69329102f38af1467b291e958a66475b5807a0_2_690x336.png, https://aws1.discourse-cdn.com/github/optimized/2X/9/9e69329102f38af1467b291e958a66475b5807a0_2_1035x504.png 1.5x, https://aws1.discourse-cdn.com/github/original/2X/9/9e69329102f38af1467b291e958a66475b5807a0.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/9/9e69329102f38af1467b291e958a66475b5807a0_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">Screenshot_2020-09-17 janreggie janreggie github io</span><span class="informations">1350×658 41.1 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>Am I doing something wrong here? Or are GitHub Actions not allowed on these repositories?</p>

**Comment by tcbyrd** [comment, marked as answer]
<p>The workflow is set to respond on a push to a branch named <code>main</code>, but your default branch is named <code>master</code>. In general the community has been <a href="https://github.com/github/renaming" rel="noopener nofollow ugc">moving away from the term <code>master</code> for the default branch</a> and both GitHub Pages and Actions now support this, so you’ll either need to change the default branch in your repository from <code>master</code> to <code>main</code> or change the action to respond on the branch named <code>master</code>.</p>

**Comment by janreggie** [comment]
<p>I… did not notice that. Thank you so much!!</p>
